Add hardhat (#3)
klaytn/sc-vt: hardhat/README.md

# Klaytn ServiceChain Token Bridge with Hardhat

> This project ports service chain token test/deploy to hardhat.

Reviewer: ```suggestion
This project ports ServiceChain token test/deploy to hardhat.
```

Author: done

## Testing on Hardhat network

> Run `npx hardhat run scripts/deploy.js`.

Reviewer: ```suggestion
Run `npx hardhat run scripts/deploy_and_transfer.js`.
```

Author: done

> Run `npx hardhat run scripts/deploy.js`.

Reviewer: How about adding instructions to install dependencies by running `npm install`?

Author: done

> Note that it deploys both bridge contracts and service chain tokens on the same network, so it only gives you an idea how the whole procedure works.

Reviewer: ```suggestion
Note that it deploys both bridge contracts and ServiceChain tokens on the same network, so it only gives you an idea how the whole procedure works.
```

Author: done

## Testing on Klaytn test network
### Prerequisites

> - Include `eth` to `RPC_API` in `kend.conf` and `kscnd.conf`. The feature requires klaytn >= v1.8.0.

Reviewer: ```suggestion
- Include `eth` to `RPC_API` in `kend.conf` and `kscnd.conf`. The feature requires klaytn >= v1.8.0. Also, Klaytn should be restarted after you modify the configuration files.
```

Author: done

- Customize `url`, `chainId`, `operator` in `hardhat.config.js` as follows:

```
mainbridge: {
  url: "http://127.0.0.1:8554",
  chainId: 1000,
  gasPrice: 25000000000,
      ...
  operator: '0x9388349e71140c1f099ca8293892ab0d1e151d4f',
},
subbridge: {
  url: "http://127.0.0.1:8555",
  chainId: 1001,
  gasPrice: 25000000000,
      ...
  operator: '0xcb5e2874276d3a96ab6331cafeb80baa6453eeb0',
},
```

> You may have to adjust `gasPrice` if gas price changes.

Reviewer 2: You should adjust `gasPrice` with the current `gasPrice` value from your network. The default `gasPrice` in the template is set with the value from the Cypress network.

- Run `subbridge.anchroing(true)` in kscn

> - The following address must have some Klay. Use faucet. Otherwise, you can change the `accounts` field with a preffered one.

Reviewer: ```suggestion
- The following address must have some Klay. Use faucet. Otherwise, you can change the `accounts` field with a preferred one.
```

Author: done

```
0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
```

### Running

> Run `./run.sh`. During execution, the following prompt will appear.

Reviewer: Same here, how about adding instructions to install dependencies by running `npm install`?

Author: done

# Klaytn ServiceChain Token Bridge with Hardhat

This project ports ServiceChain token test/deploy to hardhat.

## Testing on Hardhat network
Run the commands:

```
npm install
npx hardhat run scripts/deploy_and_transfer.js
```

Note that it deploys both bridge contracts and ServiceChain tokens on the same network, so it only gives you an idea how the whole procedure works.

## Testing on Klaytn test network
### Prerequisites
- Run `npm install`

> - Include `eth` to `RPC_API` in `kend.conf` and `kscnd.conf`. The feature requires klaytn >= v1.8.0. Also, Klaytn should be restarted after you modify the configuration files.

Reviewer: ```suggestion
- Include `eth` to `RPC_API` in `kend.conf` and `eth,subbridge` to `RPC_API` in `kscnd.conf`. The feature requires klaytn >= v1.8.0. Also, Klaytn should be restarted after you modify the configuration files.
```

To use `subbridge` API, we need to enable it first.

Author: done
